src/backend/env/mod.rs

```diff
@@ -989,26 +1004,79 @@ impl State {
         Ok(())
     }
 
-    pub fn create_invite(&mut self, principal: Principal, credits: Credits) -> Result<(), String> {
+    pub fn create_invite(
+        &mut self,
+        principal: Principal,
+        credits: Credits,
+        credits_per_user_opt: Option<Credits>,
+        realm_id: Option<RealmId>,
+    ) -> Result<(), String> {
+        let credits_per_user = credits_per_user_opt.unwrap_or(credits);
+        if credits % credits_per_user != 0 {
+            return Err("Credits per user are not a multiplier of credits".into());
+        }
         let min_credits = CONFIG.min_credits_for_inviting;
-        let user = self
-            .principal_to_user_mut(principal)
-            .ok_or("no user found")?;
-        if credits < min_credits {
+        let user = self.principal_to_user(principal).ok_or("no user found")?;
+        if credits_per_user < min_credits {
             return Err(format!(
                 "smallest invite must contain {} credits",
                 min_credits
             ));
         }
-        if user.credits() < credits {
-            return Err("not enough credits".into());
-        }
+
+        invite::validate_realm_id(self, realm_id.as_ref())?;
```

Comment:
I think realm validation is the concern of the state module, so I think `validate_realm_id` could be just an private function on `state` itself.

Reply:
Do you think any function that uses state directly should be in mod.rs ?

Reply:
Definitely not any, but realm validation sound like a useful function and can be probably used somewhere else too?

Reply:
Yes, i moved it to mod.